### PR TITLE
tracking game type

### PR DESCRIPTION
**Pull request by AndrewKoch**
Uses Bob zone to track menu state, thus tracking the game type.

### CODE REVIEW

File: heathskin/game_state.py [hunk 1]
```diff
@@ -76,13 +78,27 @@ def _get_heroes_from_entities(self):
     def _is_player_first(self):
         return self.entities.get('36').get_tag('ZONE') == 'OPPOSING PLAY (Hero)'
 
+    def set_game_type(self, new_game_type):
+        self.game_type = new_game_type
```
Comment (amckinley) [inline heathskin/game_state.py:82]: add info-level log here

File: heathskin/game_state.py [hunk 2]
```diff
@@ -76,13 +78,27 @@ def _get_heroes_from_entities(self):
...
+
     def feed_line(self, line):
-        pattern = "\[(?P<logger_name>\S+)\] (?P<log_source>\S+\(\)) - (?P<log_msg>.*)"  # noqa
-        results = re.match(pattern, line)
-        if not results:
-            return
 
-        self.parser.feed_line(**results.groupdict())
+        bob_pattern = "\[Bob\] ---(?P<log_msg>.*)---"  # noqa
```
Comment (amckinley) [inline heathskin/game_state.py]: noqa only needed for lines > 80 chars

File: heathskin/game_state.py [hunk 3]
```diff
@@ -76,13 +78,27 @@ def _get_heroes_from_entities(self):
...
+        bob_results = re.match(bob_pattern, line)
+
+        if bob_results:
+            results = bob_results.groupdict()
+            self.parser.feed_line("Bob", "BobLog", results["log_msg"])
+            # self.logger.info("bob_results: %s", results)
```
Comment (amckinley) [inline heathskin/game_state.py]: remove

File: heathskin/game_state.py [hunk 4]
```diff
@@ -109,6 +125,9 @@ def is_gameover(self):
         return game_ent and game_ent.get_tag("STATE") == "COMPLETE"
 
     def start_new_game(self):
+        # We're calling this function w/ a new session, not just a new game
+        # ie: 2015-04-16 23:47:10,604 INFO [game_universe.py:43] \
+        # Starting new session: (2, '2015-04-16T23:47:00')
         self.logger.info("Starting new game")
         self.entities = {}
         self.parser = LogParser(self)
```
Comment (amckinley) [inline heathskin/game_state.py:131]: reset game_type here as well

File: heathskin/log_parser.py [hunk 5]
```diff
@@ -57,6 +59,7 @@ def __init__(self, game_state):
         self.game_started = False
 
     def feed_line(self, logger_name, log_source, log_msg):
+        # print logger_name
```
Comment (amckinley) [inline heathskin/log_parser.py]: remove

File: heathskin/log_parser.py [hunk 6]
```diff
@@ -67,6 +70,27 @@ def feed_line(self, logger_name, log_source, log_msg):
         else:
             self.logger.debug("no parser for line %s", log_msg)
 
+    def match_bob_line(self, line):
+
+        # if "Start Spectator Game" in line:
```
Comment (amckinley) [inline heathskin/log_parser.py]: remove these lines

File: heathskin/log_parser.py [hunk 7]
```diff
@@ -339,8 +363,9 @@ def parse_zone_change_list_process_changes(self, msg):
             target_ent = self.game_state.entities[ent_data['id']]
             target_ent.update_tag("ZONE", result['dest_zone'])
 
+            # Spells being cast transition to nothing, manually set here
             if result['dest_zone'] is None:
-                self.logger.error("no zone, msg was %s", msg)
+                target_ent.update_tag("ZONE", 'CASTING SPELL')
```
Comment (amckinley) [inline heathskin/log_parser.py:366]: all other zones have underscores, right? if so, make this CASTING_SPELL

Comment (AndrewKoch) [inline heathskin/log_parser.py:366]: Other zones use spaces

File: heathskin/log_parser.py [hunk 8]
```diff
@@ -67,6 +69,23 @@ def feed_line(self, logger_name, log_source, log_msg):
...
+            "RegisterScreenEndOfGame": None,
+            "RegisterScreenBox": None,
+            "RegisterFriendChallenge": None
+            }
+
+        if line not in screen_to_game_type:
+            raise PreventableException(
+                "got unknown bob source {}".format(line))
+
+        game_type = screen_to_game_type[line]
```
Comment (amckinley) [inline heathskin/log_parser.py:86]: this will set the game_type to be "None". instead, just dont even set it

File: heathskin/game_state.py [hunk 9]
```diff
@@ -76,13 +78,28 @@ def _get_heroes_from_entities(self):
...
-        self.parser.feed_line(**results.groupdict())
+        bob_pattern = "\[Bob\] ---(?P<log_msg>.*)---"
+        bob_results = re.match(bob_pattern, line)
+
+        if bob_results:
+            results = bob_results.groupdict()
+            self.parser.feed_line("Bob", "BobLog", results["log_msg"])
+
+        else:
+            pattern = "\[(?P<logger_name>\S+)\] (?P<log_source>\S+\(\)) - (?P<log_msg>.*)"  # noqa
```
Comment (amckinley) [inline heathskin/game_state.py:96]: still doesnt need this #noqa

Comment (AndrewKoch) [inline heathskin/game_state.py:96]: This line is 96 characters

**Pull request merged**